### PR TITLE
Migrate to null safety

### PR DESCRIPTION
**Pull request by Molkars**
This pull request migrates to mime_type to null-safety and updates the version to 1.0.0-nullsafety

### CODE REVIEW

**Review by IchordeDionysos — APPROVED** [review]
@molkars awesome 😍  thanks for the PR 👍

File: pubspec.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 name: mime_type
-version: 0.3.1
+version: 1.0.0-nullsafety
```
[the remark this answers is not in the document]
Comment (IchordeDionysos) [reply pubspec.yaml:2]: Also, probably today there will a release of the Dart null-safety feature to stable ☺️

[the remark this answers is not in the document]
Comment (Molkars) [reply pubspec.yaml:2]: No problem, I suppose we could remove the beta tag from the sdk constraints as well now that 2.12 is a stable release.

[the remark this answers is not in the document]
Comment (IchordeDionysos) [reply pubspec.yaml:2]: Yes 😍

[the remark this answers is not in the document]
Comment (Molkars) [reply pubspec.yaml:2]: I suppose @mitsuoka is the one that has to update the pub.dev package registry. I'm trying to migrate a package and this is one of the last blocking packages.

**Comment by mitsuoka** [comment]
As described in [Publishing packages](https://dart.dev/tools/pub/publishing) page, the pub command currently doesn’t support publishing a new package directly to a verified publisher.
Furthermore, the temporary workaround is very cumbersome for 82 years old retired engineer with cancer and some dementia.
Will someone do me a favor to be a maintainer of mime_type pub package?
I need mime_type 0.3.2 opt-in first, then upgrade it to null safe version.

**Comment by Molkars** [comment]
> 
> 
> As described in [Publishing packages](https://dart.dev/tools/pub/publishing) page, the pub command currently doesn’t support publishing a new package directly to a verified publisher.
> Furthermore, the temporary workaround is very cumbersome for 82 years old retired engineer with cancer and some dementia.
> Will someone do me a favor to be a maintainer of mime_type pub package?
> I need mime_type 0.3.2 opt-in first, then upgrade it to null safe version.

I'm sorry about your condition. I can maintain this package; I use dart quite often and I have a couple of projects that depend on mime_type.

**Comment by mitsuoka** [comment]
Thank you for your support.

2021年3月6日(土) 9:37 Dillon Shaffer <notifications@github.com>:

**Comment by mitsuoka** [comment]
In the pull request #14 (Support null safety), Mr. Marcin Szałek is requesting me to publish the current version in pub.
I found the pub command publishing problems was resolved very recently and I can publish this version anyway. 
Should I publish this?

**Comment by Molkars** [comment]
> 
> 
> In the pull request #14 (Support null safety), Mr. Marcin Szałek is requesting me to publish the current version in pub.
> I found the pub command publishing problems was resolved very recently and I can publish this version anyway.
> Should I publish this?

You should publish this null-safety version to pub.dev. It is completely stable so it doesn't need to be published as a preview or prerelease.

**Comment by mitsuoka** [comment]
Unable to publish mime_type 1.0.0.
Still I got a message from pub.dev:
“Pub Authorized Successfully
The pub client has been successfully authorized. You may now use it to upload packages and perform other tasks.”
There is no “Allow access” line to click.

**Comment by Molkars** [comment]
That means that you have authorized your client via the commandline. run `dart pub publish` to actually push it.

**Comment by mitsuoka** [comment]
Yes, I did it using command prompt.
Funny, I was able to publish the mime_type through IDE terminal previously.
I also invited you to Github and pub package.
Thank you for your advise.

**Comment by Molkars** [comment]
> Yes, I did it using command prompt.
> Funny, I was able to publish the mime_type through IDE terminal previously.
> I also invited you to Github and pub package.
> Thank you for your advise.

No problem, your session token must've expired or something so you got logged out.

**Pull request merged**